### PR TITLE
GITOPS-477 Add back non-kuttl test execution. 

### PR DESCRIPTION
**Pull request by trdoyle81**
**What type of PR is this?**
/kind code-refactoring

**What does this PR do / why we need it**:
Non-kuttl test execution was removed in a previous [ PR ](https://github.com/redhat-developer/gitops-operator/pull/611/files) as the 'printer' package is no longer available in "sigs.k8s.io/controller-runtime/pkg/envtest/printer"

**Which issue(s) this PR fixes**:

Fixes #?
[GITOPS-4777](https://issues.redhat.com/browse/GITOPS-4777)

**How to test changes / Special notes to the reviewer**:
`make test-e2e` should now execute (without RunSpecsWithDefaultAndCustomReporters)

### CODE REVIEW

**Comment by varshab1210** [comment]
Cleared a few records

```
level=fatal msg=failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to generate asset "Platform Provisioning Check": baseDomain: Invalid value: "ocp-gitops-qe.com": the zone already has record sets for the domain of the cluster: [api.ci-op-sj8g5wyn-16175.ocp-gitops-qe.com. (A), \052.apps.ci-op-sj8g5wyn-16175.ocp-gitops-qe.com. (A)]
Installer exit with code 1

```

/retest

**Comment by varshab1210** [comment]
@trdoyle81 ci/prow/v4.13-e2e and ci/prow/v4.14-e2e have some test failure. 

/test v4.12-e2e

**Comment by svghadi** [comment]
I see the non default e2e tests suite is failing with

```
go test -p 1 -timeout 1h ./test/e2e -coverprofile cover.out -ginkgo.v
ok  	github.com/redhat-developer/gitops-operator/test/e2e	581.156s	coverage: [no statements]
go test -p 1 -timeout 1h ./test/nondefaulte2e -coverprofile cover.out -ginkgo.v
....
2024-07-25T11:52:55Z	ERROR	controller-runtime.source.EventHandler	kind must be registered to the Scheme	{"error": "no kind is registered for the type v1alpha1.NotificationsConfiguration in scheme \"k8s.io/client-go/kubernetes/scheme/register.go:80\""}
```

We probably need the same import fix as the one made in `test/e2e/suite_test.go`. - @trdoyle81

File: test/e2e/suite_test.go [hunk 1]
```diff
@@ -20,11 +20,14 @@ import (
 	"context"
 	"os"
 	"path/filepath"
+	"testing"
 	"time"
 
 	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
 	// to ensure that exec-entrypoint and run can make use of them.
+	argov1alpha1api "github.com/argoproj-labs/argocd-operator/api/v1alpha1"
 	argoapi "github.com/argoproj-labs/argocd-operator/api/v1beta1"
```
Comment (svghadi) [inline test/e2e/suite_test.go:29]:
```suggestion
```
Remove this import.

File: test/e2e/suite_test.go [hunk 2]
```diff
@@ -119,6 +129,7 @@ var _ = BeforeSuite(func() {
 	Expect(configv1.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
 	Expect(templatev1.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
 	Expect(appsv1.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
+	Expect(argoapi.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
```
Comment (svghadi) [inline test/e2e/suite_test.go:132]:
```suggestion
```

This Expect is not required. Already added previous 2 `Expect` for `argov1alpha1api` & `argov1beta1api` will register the required CR.

File: test/nondefaulte2e/suite_test.go [hunk 3]
```diff
@@ -103,6 +109,7 @@ var _ = BeforeSuite(func() {
 	Expect(configv1.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
 	Expect(templatev1.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
 	Expect(appsv1.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
+	Expect(argoapi.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
```
Comment (svghadi) [inline test/nondefaulte2e/suite_test.go:112]:
```suggestion
Expect(argov1alpha1api.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())
Expect(argov1beta1api.AddToScheme(scheme.Scheme)).NotTo(HaveOccurred())	
```

Make this change and import the required deps using 
```go
argov1alpha1api "github.com/argoproj-labs/argocd-operator/api/v1alpha1"
argov1beta1api "github.com/argoproj-labs/argocd-operator/api/v1beta1"
```
in the import section at the start of the file.

**Review by svghadi — CHANGES_REQUESTED** [review]

File: test/nondefaulte2e/suite_test.go (hunk 3 above)
Comment (trdoyle81) [reply test/nondefaulte2e/suite_test.go]: Applied

File: test/nondefaulte2e/suite_test.go [hunk 4]
```diff
@@ -21,11 +21,14 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+	"testing"
 	"time"
 
 	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
 	// to ensure that exec-entrypoint and run can make use of them.
+	argov1alpha1api "github.com/argoproj-labs/argocd-operator/api/v1alpha1"
 	argoapi "github.com/argoproj-labs/argocd-operator/api/v1beta1"
```
Comment (svghadi) [inline test/nondefaulte2e/suite_test.go:30]: Can we remove the duplicate imports?

This 
```suggestion
```

and this
https://github.com/redhat-developer/gitops-operator/blob/1f42107fc31c2ae810872bca49746036fa660642/test/nondefaulte2e/suite_test.go#L97

**Review by svghadi — COMMENTED** [review]
Thanks @trdoyle81. Seems like import changes worked. Left a small suggestion, rest all looks good.

File: test/nondefaulte2e/suite_test.go (hunk 4 above)
Comment (trdoyle81) [reply test/nondefaulte2e/suite_test.go]: Done

**Pull request closed without merging**